File: くじかつ/20211010/d.py
import sys
sys.setrecursionlimit(10**6)
input = sys.stdin.readline

s = input().strip()
abc = [0]*3

for i in s:
    if i == "a":
        abc[0] += 1
    elif i == "b":
        abc[1] += 1
    else:
        abc[2] += 1

abc.sort()
m_abc = max(abc)
if m_abc-1 <= abc[0] <= m_abc and m_abc-1 <= abc[1] <= m_abc:
    print("YES")
else:
    print("NO")

File: くじかつ/20211010/d.cpp
#include <iostream>
#include <cstdio>
#include <algorithm>
#include <cmath>
#include <vector>
#include <map>
#include <set>
#include <string>
#include <queue>
#include <stack>
using namespace std;
#define MOD 1000000007
#define INF (1 << 29)
#define EPS (1e-10)
typedef long long Int;
typedef pair<Int, Int> P;

#define max(x, y) ((x) > (y) ? (x) : (y))
#define min(x, y) ((x) < (y) ? (x) : (y))
#define rep(i, n) for (int i = 0; i < n; i++)
#define rep_each(i, s) for (char i : s)

int main()
{
    string s;
    cin >> s;
    int abc[3];
    rep(i, 3) abc[i] = 0;
    rep_each(i, s)
    {
        if (i == 'a')
        {
            abc[0]++;
        }
        else if (i == 'b')
        {
            abc[1]++;
        }
        else
        {
            abc[2]++;
        }
    }
    sort(abc, abc + 3);

    if (abc[2] - 1 <= abc[0] && abc[2] - 1 <= abc[1])
    {
        cout << "YES"
             << "\n";
    }
    else
    {
        cout << "NO"
             << "\n";
    }

    return 0;
}